res/scripts/celmi/timetables/timetable.lua

```diff
@@ -199,77 +199,78 @@ function timetable.waitingRequired(vehicle)
     local time = timetableHelper.getTime()
     local currentLine = timetableHelper.getCurrentLine(vehicle)
     local currentStop = timetableHelper.getCurrentStation(vehicle)
-    if not timetableObject[tostring(currentLine)] then return false end
-    if not timetableObject[tostring(currentLine)].stations[currentStop] then return false end
-    if not timetableObject[tostring(currentLine)].stations[currentStop].conditions then return false end
-    if not timetableObject[tostring(currentLine)].stations[currentStop].conditions.type then return false end
+    local currentLineString = tostring(currentLine)
+    if not timetableObject[currentLineString] then return false end
+    if not timetableObject[currentLineString].stations[currentStop] then return false end
+    if not timetableObject[currentLineString].stations[currentStop].conditions then return false end
+    if not timetableObject[currentLineString].stations[currentStop].conditions.type then return false end
 
-    if timetableHelper.getTimeUntilDeparture(vehicle) >= 2 then return false end
+    if timetableHelper.getTimeUntilDeparture(vehicle) >= 10 then return false end
```

Comment:
The first step for making this configurable would be to have a Global Variable to get rid of all magic numbers in the code

Reply:
Could we make the mod choose the right amount of time depending on how many vehicles are on the map?

Reply:
Good Idea!
However, we still need to figure out a formula that calculates this number.

Reply:
The effort should raise linear (correct me if I'm wrong)
So the only thing we need to do is create a map with like 1000 vehicles and find a good working number for this amount of verhicles. I think it don't have to work in 100% of the cases but 95-98% should be good in my opinion. In reallife depatures are also not allways on the right time (I know what I'm talking about). But the number shouldn't be greater than 10-15 I guess.

So it's just:
y = mx+2
x = number of vehicles
y = checking time
We need to find m.
Trying for x = 1000 to find a good working y.
I hope i finde some time for testing the numbers this evening.

Reply:
Additionally I think it would be a great performance boost to greate a new function which creates an array of vehicles with active timetables. This array can be passed to the timetable waiting required function. So it only gets called for necessary vehicles. Delays in the activevehicles array shouldn't be a problem because here we don't have such an important timing. The activevehicles function needs the ability to put in new vehicles but also delete them from the array if the timetable function will be disabled for the line.

Reply:
I was thinking the same, however, I kept it like this as disabling a timetable does not restart already stopped vehicles. It would be necessary to resume all stopped vehicles as soon as you disable the timetable.
Thinking about it, this is something that's something that should be possible

Reply:
Id suggest this PR focuses on fixing the bug quickly.
I made a separate issue for improving the vehicle check loop:
-> https://github.com/IncredibleHannes/TPF2-Timetables/issues/16

Reply:
In my savegame also 5 seconds seem to be enough to work reliable. But I think it not only depends on the number of vehicles. It depends on the number of entities (vehicles, cargo, villages). As more of them as slower the simulation gets.